transforms/helpers/options.ts

```diff
@@ -1,49 +1,98 @@
+import { inspect } from 'node:util';
 import type { ZodError } from 'zod';
 import { z } from 'zod';
 import logger from './log-helper';
 import type { RuntimeData } from './runtime-data';
+import {
+  StringArraySchema,
+  StringBooleanSchema,
+  StringFalseSchema,
+} from './schema-helper';
 
 const DEFAULT_DECORATOR_CONFIG = {
   inObjectLiterals: [],
 };
 
 const DecoratorOptionsSchema = z
   .object({
-    /** Allow-list for decorators currently applied to object literal properties that can be safely applied to class properties. */
-    inObjectLiterals: z.preprocess((arg: unknown) => {
-      return typeof arg === 'string' ? arg.split(/\s*,\s*/) : arg;
-    }, z.array(z.string())),
+    inObjectLiterals: StringArraySchema.describe(
+      'Allow-list for decorators currently applied to object literal properties that can be safely applied to class properties.'
+    ),
   })
   .partial();
 
+const DecoratorsSchema = z.preprocess(
+  (arg: unknown) => {
+    return arg === true || arg === 'true' ? DEFAULT_DECORATOR_CONFIG : arg;
+  },
+  z.union([DecoratorOptionsSchema, StringFalseSchema], {
+    errorMap: (issue, ctx) => {
+      if (issue.code === z.ZodIssueCode.invalid_union) {
+        return {
+          message: `Expected DecoratorOptions object or boolean, received ${inspect(
+            ctx.data
+          )}`,
+        };
+      }
+      return { message: ctx.defaultError };
+    },
+  })
+);
+
+const QuoteSchema = z.union([z.literal('single'), z.literal('double')], {
```

Comment:
It seems surprising to me that you need to write so many long-hand error messages when you're already using `z.union`. Admittedly I just learned about `zod`, so I may be misunderstanding how it's intended to be used.

Reply:
I really like zod, but I'm pretty disappointed with the error messages for union types. They seem to be pretty generic, along the lines of "invalid union." I thought of making a shared `errorMap` function for unions that joins the messages together for each of the types within the union, but ended up needing too many special cases for it to make sense. If we add more unions it might be worth it to abstract the union error handling further.

Reply:
Makes total sense to me.